### DISCUSSION TITLE
Running `gh api repos/my_org/my_repo/tarball/branchname` is getting a "Connection reset by peer" from codeload.github.com

### DISCUSSION THREAD

**Discussion by DiceJonSpriggs** [community/community#73281] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I have a build script which, as part of it's code performs:

```
gh api "repos/${ORGNAME}/${REPONAME}/tarball/${BRANCHNAME}" | tar xfz -
```

When this command is run, I get:

```
Get "https://codeload.github.com/${ORGNAME}/${REPONAME}/legacy.tar.gz/refs/heads/${BRANCHNAME}?token=${A_LONG_TOKEN}": read tcp 192.168.2.230:36152->140.82.121.10:443: read: connection reset by peer
```

This has been working fine until today. `gh auth status` returns:

```
github.com
  ✓ Logged in to github.com as DiceJonSpriggs (${SOMETHING})
  ✓ Git operations for github.com configured to use ${SOMEMETHOD}.
  ✓ Token: gho_************************************
  ✓ Token scopes: gist, read:org, repo, workflow
```

So, I am signed in :)

Any ideas? I've got a workaround for now, but knowing what's up would be useful! :)

**Comment by ebarzilay-godaddy** [comment]
@DiceJonSpriggs I just ran into this too.  What's your workaround?

**Reply by DiceJonSpriggs** [reply]
It's a bit pants, but I have a local clone on my machine, and I just did a clone against that. It's not great for "everyone else" though. Realistically, the tarball is "just" a clone with the .git directory removed - it's just convenient not having to do that for this specific situation, especially when I can use `gh` to access the API and pull down a tarball at a specific branch or commit.

**Reply by ebarzilay-godaddy** [reply]
Yeah, that won't do in my case...

In any case, it looks like it's a `gh` bug -- works in <= 2.32.0 and fails with >= 2.33.0

**Comment by williammartin** [comment]
Have responded in https://github.com/cli/cli/issues/8289#issuecomment-1792812601
